### PR TITLE
Move generated message to details folder

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🦟 Bug fix

## Summary

Our doxygen process filters out files in in the `detail` folder. With the `ign` to `gz` rename, we moved all of the generated message files to a `detail` subfolder. This prevented them from being processed by doxygen.

This PR instead uses a `details` (note the plural s) folder for the generated messages.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -72,7 +72,7 @@ function(gz_msgs_protoc)
     # Full path to generated header (${PROJECT_BINARY_DIR}/include/gz/msgs/foo.pb.h)
     set(output_header "${gz_msgs_protoc_OUTPUT_CPP_DIR}${proto_package_dir}/${FIL_WE}.pb.h")
     # Full path to generated detail header (${PROJECT_BINARY_DIR}/include/gz/msgs/detail/foo.pb.h)
```
Comment (caguero) [inline src/CMakeLists.txt:74]: `details` in this comment as well?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -200,7 +200,7 @@ gz_install_includes(
 
 # Install gz/msgs/detail
```
Comment (caguero) [inline src/CMakeLists.txt:201]: `details`?

File: src/CMakeLists.txt (hunk 1 above)
Comment (nkoenig) [reply src/CMakeLists.txt]: [bec5cd9](https://github.com/gazebosim/gz-msgs/pull/304/commits/bec5cd9440985441fb315f4781ef6f1b273236af)

File: src/CMakeLists.txt (hunk 2 above)
Comment (nkoenig) [reply src/CMakeLists.txt]: [bec5cd9](https://github.com/gazebosim/gz-msgs/pull/304/commits/bec5cd9440985441fb315f4781ef6f1b273236af)

**Review by caguero — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**